tests/unit/lib/DynamicArray_unittest.cc

```diff
@@ -251,10 +252,44 @@ TEST_F(CDynamicArrayFixture, append_array_bool)
 	EXPECT_EQ(wrapper_array_b->get_element(2), true);
 }
 
-TEST_F(CDynamicArrayFixture, serializable_array)
+TYPED_TEST(CDynamicArrayTest, serializable_array)
 {
-	wrapper_array_i->save_serializable_pre();
-	wrapper_array_i->load_serializable_pre();
-	wrapper_array_b->save_serializable_pre();
-	wrapper_array_b->load_serializable_pre();
+	for (int32_t i = 0; i < 5; i++)
+	{
+		this->custom_array->push_back((TypeParam)CMath::random(0, 1));
+	}
+
+	/* generate file name */
+	char filename_tmp[] = "serialization-asciiCDynamicArray.XXXXXX";
+#ifdef _WIN32
```

Comment:
There is a few places where we use temporary filenames. All of those should use the same helper function